### ISSUE TITLE
GitlabCommitStatus throws error after force kill

### ISSUE THREAD

**Issue by hwdegroot** [jenkinsci/gitlab-plugin#506]
### Context
- **Gitlab plugin version**: 1.4.4
- **Gitlab version**: 8.15.3
- **Jenkins version**: 2.40
- **Job type**: Multibranch Pipeline

 ### Logs &  Traces

```
hudson.remoting.ProxyException: com.google.inject.CreationException: Unable to create injector, see the following errors:

1) A just-in-time binding to com.dabsquared.gitlabjenkins.workflow.GitLabBuildsStep was already configured on a parent injector.
  at org.jenkinsci.plugins.workflow.steps.ContextParameterModule.configure(ContextParameterModule.java:45)

1 error
	at com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist(Errors.java:466)
	at com.google.inject.internal.InternalInjectorCreator.initializeStatically(InternalInjectorCreator.java:155)
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:107)
	at com.google.inject.internal.InjectorImpl.createChildInjector(InjectorImpl.java:226)
	at com.google.inject.internal.InjectorImpl.createChildInjector(InjectorImpl.java:233)
	at jenkins.ProxyInjector.createChildInjector(ProxyInjector.java:110)
	at org.jenkinsci.plugins.workflow.steps.AbstractStepImpl.prepareInjector(AbstractStepImpl.java:28)
	at org.jenkinsci.plugins.workflow.steps.AbstractStepImpl.start(AbstractStepImpl.java:21)
	at org.jenkinsci.plugins.workflow.cps.DSL.invokeStep(DSL.java:182)
	at org.jenkinsci.plugins.workflow.cps.DSL.invokeMethod(DSL.java:126)
	at org.jenkinsci.plugins.workflow.cps.CpsScript.invokeMethod(CpsScript.java:108)
	at groovy.lang.GroovyObject$invokeMethod.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:151)
	at org.kohsuke.groovy.sandbox.GroovyInterceptor.onMethodCall(GroovyInterceptor.java:21)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:115)
	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:149)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:146)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:123)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:123)
	at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.methodCall(SandboxInvoker.java:16)
	at WorkflowScript.run(WorkflowScript:27)
	at ___cps.transform___(Native Method)
	at com.cloudbees.groovy.cps.impl.ContinuationGroup.methodCall(ContinuationGroup.java:57)
	at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.dispatchOrArg(FunctionCallBlock.java:109)
	at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.fixArg(FunctionCallBlock.java:82)
	at sun.reflect.GeneratedMethodAccessor550.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
	at com.cloudbees.groovy.cps.impl.ClosureBlock.eval(ClosureBlock.java:46)
	at com.cloudbees.groovy.cps.Next.step(Next.java:58)
	at com.cloudbees.groovy.cps.Continuable.run0(Continuable.java:154)
	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.access$001(SandboxContinuable.java:18)
	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable$1.call(SandboxContinuable.java:33)
	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable$1.call(SandboxContinuable.java:30)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovySandbox.runInSandbox(GroovySandbox.java:108)
	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.run0(SandboxContinuable.java:30)
	at org.jenkinsci.plugins.workflow.cps.CpsThread.runNextChunk(CpsThread.java:163)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:328)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$100(CpsThreadGroup.java:80)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:240)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:228)
	at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:63)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:112)
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
```

### Problem description
When a job that uses the plugin is killed, it will try to graceful terminate steps. But when the build is `forcibly killed`, the next time the build will fail on the [gitlabCommitstatus](#logs-traces)

```
Click here to forcibly terminate running steps
Aborted by Someone
[Pipeline] }
Click here to forcibly terminate running steps
Terminating gitlabCommitStatus
Click here to forcibly kill entire build
Terminating gitlabCommitStatus
Hard kill!
```

**Comment by mtyurt** [comment]
Same problem with Gitlab plugin version 1.4.5 and Jenkins version 2.45

**Comment by segersb** [comment]
Hello,
Is there a workaround for this issue?

**Comment by omehegan** [comment]
@segersb there is a PR which fixes it, I'll try to do a release this week.

**Comment by john-forrest** [comment]
@omehegan Any news on this? [Hit same issue. Happened when restarted with saftExit, even though previous job was underway. Failed on restart. There is probably another defect there too, but subsequently hit this.]

**Comment by DieBauer** [comment]
I've ran in exactly the same issue. Is there a workaround? And when will the pull request be merged in?

**Comment by nickpetrovic** [comment]
@DieBauer i ran into this today and found there was a new [release](https://github.com/jenkinsci/gitlab-plugin/blob/master/CHANGELOG.md#151) for this plugin. try updating and see if that resolves the issue.

**Comment by petewilcock** [comment]
+1. When we see this we need to restart Jenkins once or twice to clear what seems to be bad state.

**Comment by omehegan** [comment]
This was just fixed/released in version 1.5.3 of the plugin. It should be in the update center within a few hours.
